engineio/server.go

```diff
@@ -0,0 +1,36 @@
+package engineio
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/googollee/go-socket.io/log"
+)
+
+type Options func(*Server)
+
+func OptionPingInterval(time.Duration) Options                      { return nil }
+func OptionPingTimeout(time.Duration) Options                       { return nil }
+func OptionMaxBufferSize(int) Options                               { return nil }
+func OptionLogger(logger log.Logger) Options                        { return nil }
+func OptionTransports(initial string, upgradings ...string) Options { return nil }
```

Comment:
i suggest add transport type similar as a packet type.

for example:
```go
type Tranport uint
const (
	Polling Tranport = iota
	Websocket
	SSE
)
```

Reply:
We could have predefined transport in string later.

Reply:
Done.